virttest/utils_time.py

```diff
@@ -0,0 +1,208 @@
+            return (False, session)
+
+    # 'check_tz_cmd' got a name of timezone that only used in Windows os, we
+    # need to check the whole list of timezone in Windows to get the
+    # corresponding UTC code.
+    elif os_type == "windows":
+        check_tz_cmd = "tzutil /g"
+        set_tz_cmd = 'tzutil /s "%s"'
+        list_tz_cmd = "tzutil /l"
+
+        o_guest = session.cmd_output(check_tz_cmd)
+        # guest_tzs: (timezone code, timezone name(Win))
+        guest_tzs = []
+        guest_tz = []
+        tz_list = session.cmd_output(list_tz_cmd)
+        tz_list = tz_list.splitlines()
+        for line in tz_list:
+            # empty line
+            if not line:
+                continue
+            # try to get the UTC code from the line
+            og = re.search(r'(?:\(UTC([+|-]\d{2}:\d{2})?)', line,
+                           re.M | re.I)
+            if og and og.group(1):
+                og_ = re.sub(r'(\d{2}):(\d{2})', r'\1\2', og.group(1))
+                guest_tz.append(og_)
+                continue
+            # when it is UTC time, add timezone code manually.
+            elif og:
+                guest_tz.append("+0000")
+                continue
+            og_value = re.search(r'(\S+(?:\s\S+)*)', line, re.M | re.I)
+            if og_value:
+                guest_tz.append(og_value.group(0))
+            else:
+                logging.warn("Can not get the name of timezone correctly.")
+
+            guest_tzs.append(guest_tz)
+            guest_tz = []
+
+        for value in guest_tzs:
+            # Traverse list, get timzone code by timezone name.
+            if o_guest == value[1] and (
+                    value[0] == host_tz.get('code_timezone')):
+                return (True, session)
+
+        # Windows guest's timezone not match host's timezone
+        # set windows guest's timezone with timezone name.
+        tz_name = ""
+        for value in guest_tzs:
+            if host_tz.get('code_timezone') == value[0]:
+                tz_name = value[1]
+                break
+        status = session.cmd_status(set_tz_cmd % tz_name)
+
+    if not status:
+        # boot again
+        vm.destroy()
+        vm.create(params)
+        vm.verify_alive()
+        session = vm.wait_for_login()
+        return (True, session)
+    else:
+        logging.warn("Fail to update timezone in guest")
+        return (False, session)
+
+
+def set_clocksource(expected, session=None):
+    """
+    Handle host/guest's clocksource info.
+
+    :param session: VM session or host session.
+    :param expected: Clocksource want to set to.
+
+    :return: True if succeed.
+    """
+    clocksource_dir = "/sys/devices/system/clocksource/clocksource0/"
+    clocksource_dir += "current_clocksource"
+    status = -1
+    try:
+        if session:
+            status = session.cmd_status(
+                    "echo %s > %s" % (expected, clocksource_dir), shell=True)
```

Comment:
`TypeError: cmd_status() got an unexpected keyword argument 'shell'`

`shell=True ` we use it only for process

Reply:
Thanks so much! I will update it.